share/completions/s3cmd.fish

```diff
@@ -0,0 +1,178 @@
+# s3cmd
+
+function __s3cmd_is_remote_path
+	commandline -pct | string match -r -- "^s3://.*"
+end
+
+function __s3cmd_is_valid_remote_path
+	commandline -pct | string match -r -- "^s3://.*?/.*"
+end
+
+function __prev_tokens
+	set __tokens (commandline -pco)
+	for arg in $argv
+		if echo $__tokens[-1] | string match $arg
+			return 0
+		end
+	end
+	return 1
+end
+
+function __initial_command
+	set __tokens (commandline -pco)
+	string match (count $__tokens) 1
```

Comment:
This is a bit unorthodox.

Strictly, this would be `set -q __tokens[1]; and not set -q __tokens[2]`. You could probably get by without the first check, depending on what exactly you want to do with this.

Alternatively, `test (count $__tokens) = 1`.